### ISSUE TITLE
PCL and OpenCV conflicting VTK versions

### ISSUE THREAD

**Issue by miksendk** [Homebrew/homebrew-core#73747]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
HOMEBREW_VERSION: 3.0.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8b677c0ddd5f96b5521c1b353a9ca07dfe57b80e
Last commit: 2 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 39ee99bda8fa477128a4bfb231515dea4e0d3bd0
Core tap last commit: 9 days ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 12.0 build 1200
Git: 2.29.2 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4

Unexpected dylibs:
  /usr/local/lib/libsfml-audio.2.5.1.dylib
  /usr/local/lib/libsfml-graphics.2.5.1.dylib
  /usr/local/lib/libsfml-network.2.5.1.dylib
  /usr/local/lib/libsfml-system.2.5.1.dylib
  /usr/local/lib/libsfml-window.2.5.1.dylib

HOMEBREW_VERSION: 3.0.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8b677c0ddd5f96b5521c1b353a9ca07dfe57b80e
Last commit: 2 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 39ee99bda8fa477128a4bfb231515dea4e0d3bd0
Core tap last commit: 9 days ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 12.0 build 1200
Git: 2.29.2 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4
```

### 

- [X] The `brew doctor` above contains no "Warning" lines.

### What were you trying to do (and why)?

I'm trying to build a simple project using OpenCV and PCL together for creating point clouds.

### What happened (include all command output)?

OpenCV and PCL require two different versions of VTK. For that reason, a memory error occurs (I believe). This is the output I receive after successful build:

objc[68491]: Class vtkCocoaTimer is implemented in both /usr/local/opt/vtk/lib/libvtkRenderingUI-9.0.1.dylib (0x119d9f3d0) and /usr/local/opt/vtk@8.2/lib/libvtkRenderingOpenGL2-8.2.1.dylib (0x100d6fb18). One of the two will be used. Which one is undefined.
objc[68491]: Class vtkCocoaFullScreenWindow is implemented in both /usr/local/opt/vtk/lib/libvtkRenderingOpenGL2-9.0.1.dylib (0x119cecec8) and /usr/local/opt/vtk@8.2/lib/libvtkRenderingOpenGL2-8.2.1.dylib (0x100d6fb90). One of the two will be used. Which one is undefined.
objc[68491]: Class vtkCocoaServer is implemented in both /usr/local/opt/vtk/lib/libvtkRenderingOpenGL2-9.0.1.dylib (0x119cecef0) and /usr/local/opt/vtk@8.2/lib/libvtkRenderingOpenGL2-8.2.1.dylib (0x100d6fbb8). One of the two will be used. Which one is undefined.
objc[68491]: Class vtkCocoaGLView is implemented in both /usr/local/opt/vtk/lib/libvtkRenderingOpenGL2-9.0.1.dylib (0x119cecf40) and /usr/local/opt/vtk@8.2/lib/libvtkRenderingOpenGL2-8.2.1.dylib (0x100d6fc08). One of the two will be used. Which one is undefined.

### What did you expect to happen?

Expected code to run, and the libraries to use each their version of VTK. It seems this is a problem that should be dealt with by the OpenCV formula people.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install opencv
brew install boost
brew install eigen
brew install flann
brew install vtk
brew install pcl
```

**Comment by SMillerDev** [comment]
> It seems this is a problem that should be dealt with by the OpenCV formula people.

Those don't really exist. All homebrew Formula are maintained by all homebrew maintainers. But if pcl or opencv can't work with the latest VTK version that should be reported (or reminded) upstream so that the developers can improve on that. Since I imagine every package manager will have this problem.

**Comment by miksendk** [comment]
I see. Do you have an idea for a workaround? Im unfortunately pressured on time.

**Comment by miksendk** [comment]
I see, thank you. What I don't understand is that Opencv from brew should supposedly depend on VTK 9.0 as seen on this page: https://formulae.brew.sh/formula/opencv

However my output clearly shows that OpenCV uses VTK 8.2 while PCL uses 9.0, unless im missing something.

**Comment by carlocab** [comment]
I think you have it the other way around: https://formulae.brew.sh/formula/pcl

**Comment by Bo98** [comment]
PCL uses VTK 8.2. This will be fixed to use VTK 9.0 when PCL 1.12 is released - PCL 1.11 doesn't work with VTK 9.0.

I could potentially update the `--HEAD` build of PCL to use VTK 9.0 in the meantime.

**Comment by miksendk** [comment]
Ah my bad, thank you for correcting me.

If you could do that Bo98 that would be great. How can I utilize that? Just brew upgrade pcl?

**Comment by miksendk** [comment]
Alternatively, which OpenCV version can I replace the current version with, to downgrade to VTK 8.2 support?

**Comment by Bo98** [comment]
That will affect people who rely on VTK 9. Not everyone use the PCL + OpenCV combo.

I am adjusting the `--HEAD` build of PCL here: https://github.com/Homebrew/homebrew-core/pull/73765

Once merged you can do a `brew uninstall pcl && brew install --HEAD pcl`. But be aware that:

* It's not a stable release - so there is no guarantee how well it will work.
* It will not have bottles so will need to be built from source. PCL can take as much as an hour to build.
* Future upgrades will be disabled. When PCL 1.12 stable is released, you will need to `brew uninstall pcl && brew install pcl` to get it (`brew reinstall` is insufficient for this).

**Comment by Bo98** [comment]
I've merged #73765. That's about all we can do here.
